core/src/main/java/io/questdb/cairo/CairoMetadata.java

```diff
@@ -0,0 +1,328 @@
+
+                if (columnType > -1) {
+                    String columnName = Chars.toString(name);
+                    CairoColumn column = new CairoColumn();
+
+                    LOG.debug().$("hydrating column [table=").$(token.getTableName()).$(", column=").$(columnName).I$();
+
+                    column.setName(columnName);
+                    table.upsertColumn(column);
+
+                    column.setPosition((int) (table.getColumnCount() - 1 < 0 ? 0 : table.getColumnCount() - 1));
+                    column.setType(columnType);
+
+                    if (column.getType() < 0) {
+                        // deleted
+                        continue;
+                    }
+
+                    column.setIsIndexed(TableUtils.isColumnIndexed(metaMem, writerIndex));
+                    column.setIndexBlockCapacity(TableUtils.getIndexBlockCapacity(metaMem, writerIndex));
+                    column.setIsSymbolTableStatic(true);
+                    column.setIsDedupKey(TableUtils.isColumnDedupKey(metaMem, writerIndex));
+                    column.setWriterIndex(writerIndex);
+                    column.setStableIndex(stableIndex);
+                    column.setIsDesignated(writerIndex == table.getTimestampIndex());
+                    column.setIsSequential(TableUtils.isSequential(metaMem, writerIndex));
+
+                    if (column.getIsDedupKey()) {
+                        table.setIsDedup(true);
+                    }
+
+                    if (ColumnType.isSymbol(columnType)) {
+                        LOG.debug().$("hydrating symbol metadata [table=").$(token.getTableName()).$(", column=").$(columnName).I$();
+
+                        // get column version
+                        path.trimTo(configuration.getRoot().length())
+                                .concat(table.getDirectoryName())
+                                .concat(TableUtils.COLUMN_VERSION_FILE_NAME);
+                        columnVersionReader.ofRO(configuration.getFilesFacade(),
+                                path.$());
+
+                        final long columnNameTxn = columnVersionReader.getDefaultColumnNameTxn(writerIndex);
+                        columnVersionReader.close();
+
+                        // use txn to find correct symbol entry
+                        final LPSZ offsetFileName = TableUtils.offsetFileName(
+                                path.trimTo(configuration.getRoot().length()).concat(table.getDirectoryName())
+                                , columnName, columnNameTxn);
+
+                        // initialise symbol map memory
+                        MemoryCMR offsetMem = Vm.getCMRInstance();
+                        final long offsetMemSize = SymbolMapWriter.keyToOffset(0) + Long.BYTES;
+                        offsetMem.of(configuration.getFilesFacade(), offsetFileName, offsetMemSize, offsetMemSize, MemoryTag.NATIVE_METADATA_READER);
```

Comment:
all of the "of" calls throw exceptions if they cannot open file, it must be handled along with logging